### PR TITLE
Add param to simplify loading `cvmfs::fsck`

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
Add a simple parameter to ease loading `cvmfs::fsck` via arguments.

#### This Pull Request (PR) fixes the following issues
None

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -88,6 +88,7 @@
 # @param cvmfs_repo_list Specify exactly the REPO_LIST in `defaults.local` overriding auto population.
 # @param cvmfs_alien_cache Use an alien cache
 # @param cvmfs_shared_cache Enable a shared cache
+# @param cvmfs_include_fsck Ensure the cvmfs::fsck class is included.
```
Comment (traylenator) [inline manifests/init.pp:91]:
```suggestion
# @param cvmfs_fsck Ensure the cvmfs::fsck class is included.
```
Don't think we want include in variable name as the method does not need to be exposed.

File: manifests/init.pp [hunk 2]
```diff
@@ -144,6 +145,7 @@
   Optional[String] $cvmfs_alien_cache                                 = undef,
   Optional[Enum['yes','no']] $cvmfs_shared_cache                      = undef,
   Optional[String[1]] $cvmfs_repositories                             = undef,
+  Boolean $cvmfs_include_fsck                                         = false,
```
Comment (traylenator) [inline manifests/init.pp:148]:
```suggestion
  Boolean $cvmfs_fsck                                         = false,
```

File: manifests/init.pp [hunk 3]
```diff
@@ -182,6 +184,10 @@
   contain 'cvmfs::service'
   Class['Cvmfs::Install'] -> Class['Cvmfs::Config'] ~> Class['Cvmfs::Service']
 
+  if $cvmfs_include_fsck {
```
Comment (traylenator) [inline manifests/init.pp:187]:
```suggestion
  if $cvmfs_fsck {
```

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -303,6 +303,23 @@
             it { is_expected.not_to contain_service('autofs') }
           end
 
+          context 'with cvmfs_include_fsck not set' do
+            it do
+              is_expected.not_to contain_class('cvmfs::fsck')
+            end
+          end
+
+          context 'with cvmfs_include_fsck set to true' do
```
Comment (traylenator) [inline spec/classes/init_spec.rb:312]:
```suggestion
          context 'with cvmfs_fsck set to true' do
```

File: spec/classes/init_spec.rb [hunk 5]
```diff
@@ -303,6 +303,23 @@
...
+            let(:params) do
+              { cvmfs_include_fsck: true,
```
Comment (traylenator) [inline spec/classes/init_spec.rb:314]:
```suggestion
              { cvmfs_fsck: true,
```

**Review by traylenator — CHANGES_REQUESTED** [review]
Prefer `cmvfs_fsck` over

**Comment by traylenator** [comment]
If we add that option we may as well go the whole way and make cvmfs::fsck a private interface that gets it values from the cvmfs class.

So:

```puppet
class cvmfs::fsck (                            
  $cvmfs_cache_base = $cvmfs::cvmfs_cache_base,
  $options =  $cvmfs::cvmfs_fsck_options,
  $onreboot = $cvmfs::cvmfs_fsck_onreboot,
  Optional[Boolean] $usesystemd = undef,
) inherits cvmfs {

```
As for last usesystemd this looks be completly unnecessary these days. Leave as undef.

**Comment by jcpunk** [comment]
I've attempted to resolve all requested changes.

**Review by traylenator — CHANGES_REQUESTED** [review]
Code all looks fine.

Can you update the README as it still references the cvmfs::fsck which is now not needed.
Can the @params at the top of `cvmfs::fsck` be replaced with:

# @api private

Lets make this backwards incompatible even if the old API still works.
Better that folk know that old api is gone even if it still works.

**Comment by jcpunk** [comment]
Sounds good.  I'll confess, I have no idea how to export the values in rspec to get the tests happy.....

**Comment by jcpunk** [comment]
Works for me, I've put the params (and their doc to make the analysis happy) back and dropped `assert_private`

**Comment by jcpunk** [comment]
This version look viable?

**Pull request merged**